sdk/samples/iot/provisioning/src/paho_iot_provisioning_example.c

```diff
@@ -325,131 +337,161 @@ static int get_operation_status()
-        print_az_span(response.operation_status);
-        printf("\tLast operation status: %d\n", response.status);
-        printf("\tOperation ID: ");
-        print_az_span(response.operation_id);
-        printf("\tError code: %u\n", response.registration_result.extended_error_code);
-        printf("\tError message: ");
-        print_az_span(response.registration_result.error_message);
-        printf("\tError timestamp: ");
-        print_az_span(response.registration_result.error_timestamp);
-        printf("\tError tracking ID: ");
-        print_az_span(response.registration_result.error_tracking_id);
-        if (response.retry_after_seconds > 0)
-        {
-          printf("\tRetry-after: %u seconds.", response.retry_after_seconds);
-        }
-      }
+      printf("\tRetry-after: %u seconds.", response.retry_after_seconds);
     }
+  }
+
+  return get_operation_status_free(MQTTCLIENT_SUCCESS, topic, message);
+}
+
+static az_result parse_operation_message(
+    char* topic,
+    int topic_len,
+    MQTTClient_message* message,
+    az_iot_provisioning_client_register_response* response,
+    az_iot_provisioning_client_operation_status* operation_status)
+{
+  int rc;
 
-    MQTTClient_freeMessage(&message);
-    MQTTClient_free(topic);
+  printf("Topic: %s\n", topic);
+
+  if (topic_len == 0)
+  {
+    // The length of the topic if there are one or more NULL characters embedded in topic,
+    // otherwise topic_len is 0.
+    topic_len = (int)strlen(topic);
+  }
+
+  az_span topic_span = az_span_init((uint8_t*)topic, topic_len);
+  az_span message_span = az_span_init((uint8_t*)message->payload, message->payloadlen);
+
+  // Parse the incoming message and payload
+  if (az_failed(
+          rc = az_iot_provisioning_client_parse_received_topic_and_payload(
+              &provisioning_client, topic_span, message_span, response)))
+  {
+    printf("Message from unknown topic, az_result return code %0x4\n", rc);
+    return rc;
+  }
+  printf("Received payload:\n");
+  print_az_span(message_span);
+  printf("Response status is %d.\n", response->status);
+
+  // Parse the operation status from a string to an enum
+  if (az_failed(rc = az_iot_provisioning_client_parse_operation_status(response, operation_status)))
+  {
+    printf("Failed to parse operation_status, az_result return code %0x4\n", rc);
+    return rc;
+  }
+
+  return AZ_OK;
+}
+
+static int send_operation_query_message(az_iot_provisioning_client_register_response response)
+{
+  int rc;
+
+  printf("sending operation query message\n");
+
+  // Get the topic to send the query message
+  if (az_failed(
+          rc = az_iot_provisioning_client_query_status_get_publish_topic(
+              &provisioning_client, &response, query_topic, sizeof(query_topic), NULL)))
+  {
+    printf("Unable to get query status publish topic, az_result return code %04x\n", rc);
+    return rc;
+  }
+
+  // IMPORTANT: Wait the recommended retry-after number of seconds before query
```

Comment:
Same as above (here and throughout the file): please end comment sentences in "."

Reply:
same as above.